code/modules/projectiles/guns/energy.dm

```diff
@@ -33,6 +33,10 @@
 	var/recharge_multiplier = 1
 	var/charge_tick = 0
 
+	/// External power sources for the gun. Checked by get_external_power_supply()
+	var/obj/item/rig_module/recharger/rigcharger
+	var/obj/item/recharger_backpack/backcharger
```

Comment:
You don't need these two to be separate - either unify these variables as a single item then use get_cell() to get the cell from it, or make the rigcharger just be a backpack charger but in the rig.